app/services/stepfunctions/RegularContributionsClient.scala

```diff
@@ -136,3 +126,41 @@ class RegularContributionsClient(
     underlying.status.map(_.getStatus == "ACTIVE").getOrElse(false)
 
 }
+
+object StepFunctionExecutionStatus {
+
+  case class FinishedState(name: String, eventDetails: StateExitedEventDetails)
+
+  def checkoutStatus(detailedHistory: List[Try[StateExitedEventDetails]], stateWrapper: StateWrapper, trackingUri: String): StatusResponse = {
+
+    val finishedStates: List[Try[FinishedState]] = detailedHistory.map { stateAttempt =>
```

Comment:
I still think this is unnecessary complex, but at least it could be simplified to
```scala
detailedHistory.map(_.map(attempt => FinishedState(attempt.getName, attempt)))
```
😄

Reply:
Sorry, I didn't notice this until after merging - I will have a look at simplifying in a follow up PR